internal/versions/versions_test.go

```diff
@@ -592,6 +592,62 @@ var _ = Describe("list versions", func() {
 		})
 	})
 
+	Context("ValidateReleaseImageForRHCOS", func() {
+		BeforeEach(func() {
+			osImages = &defaultOsImages
+			h, err = NewHandler(logger, mockRelease, versions, *osImages, models.ReleaseImages{
+				&models.ReleaseImage{
+					CPUArchitecture:  swag.String(common.MultiCPUArchitecture),
+					CPUArchitectures: []string{common.X86CPUArchitecture, common.ARM64CPUArchitecture},
+					OpenshiftVersion: swag.String("4.11.1"),
+					URL:              swag.String("release_4.11.1"),
+					Default:          false,
+					Version:          swag.String("4.11.1-chocobomb-for-test"),
+				},
+				&models.ReleaseImage{
+					CPUArchitecture:  swag.String(common.MultiCPUArchitecture),
+					CPUArchitectures: []string{common.X86CPUArchitecture, common.ARM64CPUArchitecture},
+					OpenshiftVersion: swag.String("4.12"),
+					URL:              swag.String("release_4.12"),
+					Default:          false,
+					Version:          swag.String("4.12"),
+				},
+			}, nil, "", authzHandler)
+			Expect(err).ShouldNot(HaveOccurred())
+		})
+
+		It("validates successfuly using exact match", func() {
+			err = h.ValidateReleaseImageForRHCOS("4.11.1", common.X86CPUArchitecture)
+			Expect(err).ShouldNot(HaveOccurred())
+		})
+		It("validates successfuly using major.minor", func() {
+			err = h.ValidateReleaseImageForRHCOS("4.11", common.X86CPUArchitecture)
+			Expect(err).ShouldNot(HaveOccurred())
+		})
+		It("validates successfuly using major.minor using default architecture", func() {
+			err = h.ValidateReleaseImageForRHCOS("4.11", "")
+			Expect(err).ShouldNot(HaveOccurred())
+		})
+		It("validates successfuly using major.minor-something", func() {
```

Comment:
Did you mean this name to by `major.minor.patch-something` or did we actually want to test `4.12-chocobomb`?

Reply:
Honestly, not a difference because the core of the test is that both `4.12.2-chocobomb` and `4.12-chocobomb` fold into `4.12` and we check if we have release image for `4.12` available

Reply:
Let me fix the name then. `4.12-chocobomb` is not natural and we should not encounter it anywhere in the wild